VolleyManagement.UI/Infrastructure/CaptchaManager.cs

```diff
@@ -5,38 +5,57 @@
     using System.Web.Configuration;
     using Contracts;
     using Newtonsoft.Json.Linq;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using System;
 
     /// <summary>
     /// Represents an Implementation of ICaptchaManager interface.
     /// </summary>
     public class CaptchaManager : ICaptchaManager
     {
-        private const string SECRET_KEY = "RecaptchaSecretKey";
-        private const string CaptchaResponse = "g-recaptcha-response";
-        private const string CaptchaPath = "https://www.google.com/recaptcha/api/siteverify?secret={0}&response={1}";
-
+        private const string CONFIG_KEY_RECAPTCHA_SECERET = "RecaptchaSecretKey";
+        private const string RECAPTCHA_VERIFY_URL_FORMAT = "https://www.google.com/recaptcha/api/siteverify?secret={0}&response={1}";
+        private const string RECAPTCHA_JSON_TOKEN_SUCCESS = "success";
+        
         /// <summary>
         /// Method, that verifies if captcha is valid
         /// </summary>
-        /// <param name="response">Response of Captcha.</param>
+        /// <param name="userResponseToken">Response of Captcha.</param>
         /// <returns>Captcha result</returns>
-        public bool IsFormSubmit(string response)
+        public async Task<bool> ValidateUserCaptcha(string userResponseToken)
         {
-            // ToDo: Refactor: excpetion handling, naming
-            bool status = false;
-            string secretKey = WebConfigurationManager.AppSettings[SECRET_KEY];
-            using (var client = new WebClient())
+            var isCaptchaValid = false;
+            using (var client = new HttpClient())
             {
-                var result = client.DownloadString(
-                    string.Format(
-                        CaptchaPath,
-                        secretKey,
-                        response));
-                var obj = JObject.Parse(result);
-                status = (bool)obj.SelectToken("success");
+                try
+                {
+                    var jsonResultStr = await client.GetStringAsync(GetCapchaRequestUrl(userResponseToken));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call the overload that takes a 'System.Uri' as an argument instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS4005)